File: src/models/session-handle.ts
import type { Fetcher } from "@literate.ink/utilities";
import type { Queue } from "~/api/private/queue";
import type { SessionInformation } from "./session-information";
import type { UserParameters } from "./user-parameters";
import type { InstanceParameters } from "./instance-parameters";
import { UserResource } from "./user-resource";

export type SessionHandle = {
  /**
   * Equivalent of a PRONOTE session.
   * Contains metadata, AES keys, RSA modulus, and more.
   */
  information: SessionInformation
  instance: InstanceParameters
  user: UserParameters

  /**
   * Can be updated with the `pronote.use(number | UserResource)` method.
   * @default user.resources[0] // first resource from the user
   */
  userResource: UserResource

  readonly queue: Queue
  readonly fetcher: Fetcher

  presence: null | ReturnType<typeof setInterval>
};

File: library/src/commonMain/kotlin/models/SessionHandle.kt
package models

import api.private.Queue
import io.ktor.client.*

/**
 * Equivalent of a PRONOTE session.
 * Contains metadata, AES keys, RSA modulus, and more.
 */
data class SessionHandle(
    val serverURL: String,
    var information: SessionInformation,
    var instance: InstanceParameters,
    var user: UserParameters,

    val client: HttpClient = HttpClient()
)